File: 2025/dec03.cpp
#include"lib.h"

void p1(void){
    string s;
    long long sun=0;
    while(cin>>s){
        int mnum=0;
        for(int i=0;i<s.size();++i){
            for(int j=i+1;j<s.size();++j){
                int num=10*(s[i]-'0')+(s[j]-'0');
                mnum=max(mnum,num);
            }
        }
        sun+=mnum;
    }
    cout<<sun; // sol: 17074
}
int main(...){
    freopen("date.in","r",stdin);
    freopen("date.out","w",stdout);
    p1();
    return 0;
}

File: 2025/dec03.py
lines=[]
with open("date.in","r") as fileIn:
    lines=fileIn.readlines()
sum=0

# do multithreading on this shit
def do_line(line):
    line=line[:-1]
    mnum=int(0)
    for i in range(2**len(line)):
        if i.bit_count()!=12:continue
        p10=100000000000
        num=0
        for bit in range(len(line)):
            if (2**bit)&i!=0:
                num+=p10*int(line[bit])
                p10//=10
        mnum=max(mnum,num)
    return mnum

for line in lines:
    sum+=do_line(line)

with open("date.out","w") as fileOut:
    fileOut.write(str(sum))